src/types.rs

```diff
@@ -121,6 +131,77 @@ impl ResponseParam {
         }
     }
 }
+#[derive(Deserialize, Debug, Serialize)]
+pub struct EventRequest {
+    pub tx_hash: DisplayBytes,
+    pub network: String,
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct TxLog {
+    pub address: DisplayBytes,
+    pub data: DisplayBytes,
+    pub index: String,
+    pub topics: [Option<Hash>; 4],
+}
+
+#[derive(Deserialize, Debug, Serialize)]
+pub struct EventResponse {
+    pub events: Vec<DecodedEvent>,
+}
+
+impl EventResponse {
+    pub fn new(events: Vec<(Event, Log, String)>) -> EventResponse {
```

Comment:
why not just create `DecodedEvent` right in the `process_events`? why bother with this tuple? and accept here just a vec of right type